### ISSUE TITLE
Update development documentation

### ISSUE THREAD

**Issue by z4y4ts** [raccoongang/xblock-video#173]
[Reported](https://github.com/raccoongang/xblock-video/pull/169#issuecomment-288306588) by @bryanlandia:

* `README.md `  https://github.com/raccoongang/xblock-video/blob/dev/README.md#installation
Refers to `make deps` and  `deps` is not a valid target in the `Makefile`.  

```
edxapp@precise64:~/venvs/edxapp/src/video-xblock$ make deps
make: *** No rule to make target `deps'.  Stop.
```
It is also missing a `bower install` to get the standard (non-test) dependencies.  Am I missing something?

* There seems to be a version incompatibility issue with the dev dependencies in `package.json`. 
```
edxapp@precise64:~/venvs/edxapp/src/video-xblock$ npm install
npm WARN package.json @ No repository field.
npm WARN engine eslint@3.18.0: wanted: {"node":">=4"} (current: {"node":"0.10.37","npm":"1.4.28"})
npm WARN engine shelljs@0.7.7: wanted: {"node":">=0.11.0","iojs":"*"} (current: {"node":"0.10.37","npm":"1.4.28"})
npm WARN engine strip-bom@3.0.0: wanted: {"node":">=4"} (current: {"node":"0.10.37","npm":"1.4.28"})
npm WARN engine esprima@3.1.3: wanted: {"node":">=4"} (current: {"node":"0.10.37","npm":"1.4.28"})
npm WARN engine string-width@2.0.0: wanted: {"node":">=4"} (current: {"node":"0.10.37","npm":"1.4.28"})
npm WARN engine is-fullwidth-code-point@2.0.0: wanted: {"node":">=4"} (current: {"node":"0.10.37","npm":"1.4.28"})
npm WARN engine co@4.6.0: wanted: {"iojs":">= 1.0.0","node":">= 0.12.0"} (current: {"node":"0.10.37","npm":"1.4.28"})
npm WARN engine string-width@2.0.0: wanted: {"node":">=4"} (current: {"node":"0.10.37","npm":"1.4.28"})
npm WARN engine esprima@3.1.3: wanted: {"node":">=4"} (current: {"node":"0.10.37","npm":"1.4.28"})
npm WARN engine is-fullwidth-code-point@2.0.0: wanted: {"node":">=4"} (current: {"node":"0.10.37","npm":"1.4.28"})
npm WARN engine co@4.6.0: wanted: {"iojs":">= 1.0.0","node":">= 0.12.0"} (current: {"node":"0.10.37","npm":"1.4.28"})
npm ERR! peerinvalid The package eslint does not satisfy its siblings' peerDependencies requirements!
npm ERR! peerinvalid Peer eslint-config-airbnb-base@3.0.1 wants eslint@^2.9.0
npm ERR! peerinvalid Peer eslint-plugin-dollar-sign@1.0.0 wants eslint@>=1.0.0
npm ERR! peerinvalid Peer eslint-plugin-import@1.16.0 wants eslint@2.x - 3.x

npm ERR! System Linux 3.2.0-23-generic
npm ERR! command "/usr/bin/node" "/usr/bin/npm" "install"
npm ERR! cwd /edx/app/edxapp/venvs/edxapp/src/video-xblock
npm ERR! node -v v0.10.37
npm ERR! npm -v 1.4.28
npm ERR! code EPEERINVALID
npm ERR! 
npm ERR! Additional logging details can be found in:
npm ERR!     /edx/app/edxapp/venvs/edxapp/src/video-xblock/npm-debug.log
npm ERR! not ok code 0
```

**Comment by z4y4ts** [comment]
Hi @bryanlandia 

Two things here.

First, you need to upgrade nodejs to *v4* or higher. [Here's official docs](https://docs.npmjs.com/getting-started/installing-node#updating-npm) or [guide for Ubuntu](http://askubuntu.com/a/480642).

And second, you're right there's no `deps` make target anymore. Instead you need to run `make tools deps-test`, which will install development tools and dependencies required for tests.

Please let me know if that works for you. I'll update README.md.

**Comment by bryanlandia** [comment]
@z4y4ts - Thanks!  

Please add to the documentation that Node.js >4 is required.  Our sponsoring client is on Dogwood currently and the installed node.js package is the 0.10.x  from whenever node was installed via the old `chrislea/node.js` ppa.  

6.x is the latest major release series supported by Ubuntu Precise and therefore Dogwood.  I followed the following instructions from here:  https://nodejs.org/en/download/package-manager/#debian-and-ubuntu-based-linux-distributions  in order to update my Node/nodejs versions.

```
curl -sL https://deb.nodesource.com/setup_6.x | sudo -E bash -
sudo apt-get install -y nodejs
```

This upgrade node.js to `v6.10.1`.  Please add these instructions for Dogwood to the README.  

After that I was able to run `make tools` but `make deps-test` failed because `bower` was not added to my path after it was `npm install`ed.  I had to manually do `node_modules/bower/bin/bower install`, and that worked.

End users also won't necessarily want to install the testing packages.  The bower dependencies necessary to run the XBlock are only in `dept-test` unless I am missing something.  Can you make a `deps` target again, so that it is possible to install all end-user requirements without needing the testing requirements?  Also, the test dependencies specify the egg `common.lib.xmodule` from Eucalyptus, so that would create a problem for Dogwood installation.

Also, it looks like you have to run the `deps-test` to run `bower install`, which is needed to install the end-user dependencies.  

If you can refactor this so that testing dependencies aren't needed that would be great.  If it needs to wait until the next release, that's OK.  

I apologize for my delay working through this.

**Comment by z4y4ts** [comment]
@bryanlandia thanks for the detailed feedback!

Regarding `npm` and `bower` dependencies. It's totally valid point that these dependencies are unnecessary for a target machine. We were going to re-bundle external JavaScript and CSS into a `vendor` subfolder so that there's no need for an additional step after Xblock has been installed.

That's going to be part of the next release. We've been postponing this for so long to avoid additional packaging step during active development. But it looks like we've reached the point when we have to add this step to not overcomplicate XBlock users' life.

Regarding the `common.lib.xmodule` dependency and `test_requirements.txt` in general. They're supposed to be needed only for the testing on TravisCI. Edx-platform ships `common.lib.xmodule`, so there's no need to install it on an edx instance.

So in short, we're going to remove a need for testing dependencies on a working edx instance in the next release.
